ci/infra/testrunner/tests/test_node_reboot.py

```diff
@@ -68,4 +51,4 @@ def test_hard_reboot(deployment, platform, skuba, kubectl, node_type, node_numbe
 
     wait_for_node_reboot(node_type, node_number, platform)
     check_nodes_ready(kubectl)
-    check_pods_ready(kubectl)
+    kubectl.run_kubectl('wait --timeout=5m --for=condition=Ready pods --all --namespace=kube-system  --field-selector=status.phase!=Succeeded')
```

Comment:
The `--timeout` option for kubectl here can be replaced by `0` and wrap the `run_kubectl` with the `wait` function to make wait logic easier to understand and unify how this is done in other tests.